### PR TITLE
Do not use candidate if it is nil #3024

### PR DESCRIPTION
**Pull request by rafatower**
@Kartones can you please review?

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3295/))

File: services/importer/lib/importer/namedplaces_guesser.rb [hunk 1]
```diff
@@ -45,7 +45,11 @@ def run?
         def country_column
           return @country_column if defined?(@country_column)
           candidate = text_columns.sort{|a,b| country_proportion(a) <=> country_proportion(b)}.last
-          @country_column = (country_proportion(candidate) > @guesser.threshold) ? candidate : nil
+          @country_column = if candidate
```
Comment (Kartones) [inline services/importer/lib/importer/namedplaces_guesser.rb]: I really dislike if-assignments (outside ternary operators). But is personal taste.

I'd use
```
@country_column = (!candidate.nil? && country_proportion(candidate) > @guesser.threshold) ? candidate : nil
```

File: services/importer/lib/importer/namedplaces_guesser.rb [hunk 2]
```diff
@@ -65,7 +69,11 @@ def country_proportion(column)
         def guess_with_country_column
           candidate_columns = text_columns.reject{|c| c == country_column}
           candidate = candidate_columns.sort{|a,b| proportion(a) <=> proportion(b)}.last
-          @column = (proportion(candidate) > @guesser.threshold) ? candidate : nil
+          @column = if candidate
+                      (proportion(candidate) > @guesser.threshold) ? candidate : nil
```
Comment (Kartones) [inline services/importer/lib/importer/namedplaces_guesser.rb]: similar here :grin:

**Comment by Kartones** [comment]
:+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3304/))

**Pull request merged**